coursefinder/models.py

```diff
@@ -145,17 +147,67 @@ class CourseFinderResults(Page):
     ]
 
 
-class CourseSearch:
+class BaseSearch:
+
+    def __init__(self, page, count):
+        self.page = int(page)
+        self.count = int(count)
+        self.offset = self.count * (self.page - 1)
+        self.total_courses = None
+        self.total_institutions = None
+        self.results = None
+
+    @property
+    def show_previous_icon(self):
+        return True if self.pages_to_left else False
+
+    @property
+    def previous_page(self):
+        return self.page - 1
+
+    @property
+    def pages_to_left(self):
+        if self.page == 1:
+            return []
+        elif self.page == self.total_page_count and self.total_page_count != 2:
+            return [self.page - 2, self.previous_page]
+        else:
+            return [self.previous_page]
```

Comment:
As the elif isn't required because of the return perhaps this would be easier to read?
```suggestion
           if self.page == 1:
               return []
           if self.page == self.total_page_count and self.total_page_count != 2:
               return [self.page - 2, self.previous_page]
           return [self.previous_page]
```

Reply:
Agree this is nicer. 👍